File: typescipt/src/language/expressions/functions/monthsFunction.ts


export class MonthsFunction extends EndStartDateFunction {
   public const string Name = `MONTHS`;

   protected override string FunctionName => Name;

   constructor(endDateExpression: Expression, startDateExpression: Expression, reference: SourceReference)
     : base(endDateExpression, startDateExpression, reference) {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression endDateExpression,
     Expression startDateExpression) {
     return new MonthsFunction(endDateExpression, startDateExpression, reference);
   }
}

File: src/Lexy.Compiler/Language/Expressions/Functions/MonthsFunction.cs
namespace Lexy.Compiler.Language.Expressions.Functions;

public class MonthsFunction : EndStartDateFunction
{
    public const string Name = "MONTHS";

    private MonthsFunction(Expression endDateExpression, Expression startDateExpression, ExpressionSource source)
        : base(Name, endDateExpression, startDateExpression, source)
    {
    }

    public static FunctionCallExpression Create(ExpressionSource source, Expression endDateExpression,
        Expression startDateExpression)
    {
        return new MonthsFunction(endDateExpression, startDateExpression, source);
    }
}